core/xchainmg_net.go

```diff
@@ -390,3 +400,75 @@ func (xm *XChainMG) handleGetRPCPort(msg *xuper_p2p.XuperMessage) (*xuper_p2p.Xu
 	}
 	return xuper_p2p.NewXuperMessage(xuper_p2p.XuperMsgVersion2, "", msg.GetHeader().GetLogid(), xuper_p2p.XuperMessage_GET_RPC_PORT_RES, []byte(":"+port), xuper_p2p.XuperMessage_NONE)
 }
+
+// handleGetAuthentication callback function for handling identity authentication
+func (xm *XChainMG) handleGetAuthentication(msg *xuper_p2p.XuperMessage) (*xuper_p2p.XuperMessage, error) {
+	logid := msg.Header.Logid
+	auths := &pb.IdentityAuths{}
+	errRes := errorHandleGetAuthenMsg(logid)
+	err := proto.Unmarshal(msg.Data.MsgInfo, auths)
+	if err != nil {
+		xm.Log.Error("handleGetAuthentication unmarshal msg error", "error", err.Error())
+		return errRes, errors.New("unmarshal msg error")
+	}
+	xm.Log.Trace("Start to handleGetAuthentication", "logid", logid, "authsrequest", auths)
+
+	addrs := make([]string, 0, len(auths.Auth))
+	for _, v := range auths.Auth {
+		cryptoClient, err := crypto_client.CreateCryptoClientFromJSONPublicKey(v.Pubkey)
+		if err != nil {
+			xm.Log.Error("handleGetAuthentication Create crypto client error", "error", err.Error())
+			return errRes, errors.New("handleGetAuthentication Create crypto client error")
+		}
+
+		publicKey, err := cryptoClient.GetEcdsaPublicKeyFromJSON(v.Pubkey)
+		if err != nil {
+			xm.Log.Error("handleGetAuthentication GetEcdsaPublicKeyFromJSON error", "error", err.Error())
+			return errRes, err
+		}
+
+		isMatch, _ := cryptoClient.VerifyAddressUsingPublicKey(v.Addr, publicKey)
+		if !isMatch {
+			xm.Log.Error("handleGetAuthentication address and public key not match")
+			return errRes, errors.New("handleGetAuthentication address and public key not match")
+		}
+
+		tsNow := time.Now().Unix()
+		tsPast, err := strconv.ParseInt(v.Timestamp, 10, 64)
+		if err != nil {
+			xm.Log.Error("handleGetAuthentication timestamp fmt error")
+			return errRes, errors.New("handleGetAuthentication timestamp fmt error")
+		}
+
+		if tsNow-tsPast >= config.DefautltAuthTimeout {
```

Comment:
We usually use timers to make sure whether timeout. And the client always have a timer. Does the server need one?